### PR TITLE
Add withDecay and ScrollableView example

### PR DESCRIPTION
**Pull request by terrysahaidak**
## Description

This PR implements withDecay animation as well as adds Scrollable View example - naїve implementation of ScrollView.

## Changes

Add withDecay animation, withDecay docs and ScrollableView example

## Screenshots / GIFs

Запис екрана  о 17 29 42

### CODE REVIEW

**Comment by wcandillon** [comment]
I just tried with new version and it works well. I like the API. Sometimes when interrupting/restarting the decay a lot, it can feel a bit janky, it might be because of the emulator.

**Comment by dsznajder** [comment]
I would like to confirm, that this example & implemetation works great!. Thanks @terrysahaidak, it was very helpful for me ;)

File: src/reanimated2/animations.js [hunk 1]
```diff
@@ -176,6 +176,77 @@ export function withSpring(toValue, userConfig, callback) {
...
+  }
+
+  return {
+    animation: decay,
+    start,
+    velocity: config.velocity || 0,
+    callback,
+  };
+}
+
```
Comment (Szymon20000) [inline src/reanimated2/animations.js:249]:
```suggestion
export function withDecay(userConfig, callback) {
  'worklet';
  return defineAnimation(0, () => {
    'worklet';
    const config = {
      deceleration: 0.998,
    };
    if (userConfig) {
      Object.keys(userConfig).forEach((key) => (config[key] = userConfig[key]));
    }

    const VELOCITY_EPS = 5;

    function decay(animation, now) {
      const { lastTimestamp, initialVelocity, current, velocity } = animation;

      const deltaTime = Math.min(now - lastTimestamp, 64);
      animation.lastTimestamp = now;

      const kv = Math.pow(config.deceleration, deltaTime);
      const kx = (config.deceleration * (1 - kv)) / (1 - config.deceleration);

      const v0 = velocity / 1000;
      const v = v0 * kv * 1000;
      const x = current + v0 * kx;

      animation.current = x;
      animation.velocity = v;

      let toValueIsReached = null;

      if (Array.isArray(config.clamp)) {
        if (initialVelocity < 0 && animation.current <= config.clamp[0]) {
          toValueIsReached = config.clamp[0];
        } else if (
          initialVelocity > 0 &&
          animation.current >= config.clamp[1]
        ) {
          toValueIsReached = config.clamp[1];
        }
      }

      if (Math.abs(v) < VELOCITY_EPS || toValueIsReached !== null) {
        if (toValueIsReached !== null) {
          animation.current = toValueIsReached;
        }

        return true;
      }
    }

    function start(animation, value, now) {
      animation.current = value;
      animation.lastTimestamp = now;
      animation.initialVelocity = config.velocity;
    }

    return {
      animation: decay,
      start,
      velocity: config.velocity || 0,
      callback,
    };
  });
}

```

File: docs/docs/api/withDecay.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+### Arguments
+
+#### `options` [object]
+
+Object containing animation configuration.
+Allowed parameters are listed below:
+
+| Options  | Default | Description                                  |
+| -------- | ------- | -------------------------------------------- |
+| velocity | 0       |                                              |
```
Comment (Szymon20000) [inline docs/docs/api/withDecay.md:18]: I think we can also change the deceleration. Am I right?

File: Example/src/ScrollableViewExample.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  useAnimatedGestureHandler,
+  useAnimatedStyle,
+  useSharedValue,
+  withDecay,
+  withTiming,
+} from 'react-native-reanimated';
+import { View, Dimensions } from 'react-native';
+import { PanGestureHandler } from 'react-native-gesture-handler';
+import { Header } from 'react-navigation-stack';
+import { Easing } from '../../src/reanimated2/Easing';
```
Comment (Szymon20000) [inline Example/src/ScrollableViewExample.js:12]:
```suggestion
import { Easing } from 'react-native-reanimated';
```

File: Example/src/ScrollableViewExample.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        },
+      ],
+    };
+  });
+
+  return (
+    <View style={{ flex: 1 }}>
+      <PanGestureHandler
+        onGestureEvent={handler}
+        onHandlerStateChange={handler}>
```
Comment (Szymon20000) [inline Example/src/ScrollableViewExample.js:96]:
```suggestion
        onGestureEvent={handler}>
```

File: docs/docs/api/withDecay.md (hunk 2 above)
Comment (terrysahaidak) [reply docs/docs/api/withDecay.md:18]: Yeah, you're right.

**Pull request merged**